bin/modules/birthday/birthday_module.dart

```diff
@@ -0,0 +1,134 @@
+import 'dart:async';
+
+import 'package:cron/cron.dart';
+import 'package:hive/hive.dart';
+import 'package:nyxx/nyxx.dart';
+import 'package:nyxx_interactions/nyxx_interactions.dart';
+
+import '../bot_module.dart';
+import 'hive_constants.dart';
+
+part 'commands.dart';
+
+class BirthdayModule extends BotModule {
+  static final BirthdayModule _singleton = BirthdayModule._internal();
+
+  bool _isInitialized = false;
+
+  late INyxxWebsocket _bot;
+
+  final Cron _birthdayCron = Cron();
+
+  factory BirthdayModule() {
+    return _singleton;
+  }
+  BirthdayModule._internal();
+
+  @override
+  void init(INyxxWebsocket bot) {
+    _bot = bot;
+
+    // Set up Hive for local data storage
+    Hive.init('/data');
+    Hive.openBox(HiveConstants.birthdayBox);
+
+    _birthdayCron.schedule(Schedule.parse('0 9 * * *'), () {
+      _checkBirthdays();
+    });
+
+    _isInitialized = true;
+  }
+
+  void _checkBirthdays() {
+    if (!_isInitialized) {
+      print('Birthday module not initialized!');
+      throw Exception('Birthday module not initialized!');
+    }
+    var box = Hive.box(HiveConstants.birthdayBox);
+
+    Map<dynamic, dynamic> listOfUsers =
+        box.get(HiveConstants.birthdayUserList, defaultValue: {});
```

Comment:
Hive is kind of nice but also disgusting with these things (Have had problems with the type before), is it possible to use final instead of map?